athena-redis/src/main/java/com/amazonaws/athena/connectors/redis/lettuce/RedisConnectionFactory.java

```diff
@@ -38,19 +38,26 @@
   private static final int CONNECTION_TIMEOUT_MS = 2_000;
   private final Map<String, RedisConnectionWrapper<String, String>> clientCache = new HashMap<>();
 
-  public synchronized RedisConnectionWrapper<String, String> getOrCreateConn(String conStr, boolean sslEnabled,
-                                                                             boolean isCluster)
+  public synchronized RedisConnectionWrapper<String, String> getOrCreateConn(String conStr,
+                                                                             boolean sslEnabled,
+                                                                             boolean isCluster,
+                                                                             String dbNumber)
   {
     String conKey = conStr + sslEnabled + isCluster;
+    if (dbNumber != null && !"null".equals(dbNumber)) {
```

Comment:
Similarly relates -- if clustered mode just **doesn't support dbnum at all** then this is correct. if it's default 0, we should emulate that behavior instead.

Reply:
hmm i see. if user has multiple tables to the same clustered instances but with the same db number, we should treat them all as the same connection

Reply:
will make that change